workflow/controller/operator.go

```diff
@@ -2186,3 +2205,64 @@ func (woc *wfOperationCtx) runOnExitNode(parentName, templateRef, boundaryID str
 	}
 	return false, nil, nil
 }
+
+func (woc *wfOperationCtx) createPDBResource() error {
+
+	if woc.wf.Spec.PodDisruptionBudget == nil {
+		return nil
+	}
+
+	pdb, err := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Get(woc.wf.Name, metav1.GetOptions{})
+	if err != nil && !apierr.IsNotFound(err) {
+		return err
+	}
+	if pdb != nil && pdb.Name != "" {
+		return nil
+	}
+
+	if woc.wf.Spec.PodDisruptionBudget.Selector == nil {
+		woc.wf.Spec.PodDisruptionBudget.Selector = &metav1.LabelSelector{
```

Comment:
I don't think you should patch existing objects like this

Reply:
you are correct. it should not do that.